packages/react-native/Libraries/Components/Pressable/Pressable.js

```diff
@@ -240,6 +240,10 @@ function Pressable(props: Props, forwardedRef): React.Node {
 
   const android_rippleConfig = useAndroidRippleForView(android_ripple, viewRef);
 
+  const isStyleAFunction = typeof style === 'function';
```

Comment:
I would personally omit aliasing typeof into separate const in favor of readability. So something like this:
```
const shouldUpdateState = typeof style === 'function' || typeof children === 'function';
```
I'd expect that JS engine make subsequent `typeof` calls free. Unless that's not the case in e.g. JIT-less hermes?

Reply:
then we would have to call `typeof style === 'function';` once more which I wanted to avoid as imo is less readable, typo prone and doesn't show the relation between the `styles`/`children` functions and  `shouldUpdateState` variable

Reply:
> doesn't show the relation between the styles/children functions and shouldUpdateState variable

This relation can be neglected here: https://github.com/facebook/react-native/pull/44615/files#diff-768887e6132bb8207503a7a5d74d02de19f165eb043b82853b3a8192b3e26c06L343 and here: https://github.com/facebook/react-native/pull/44615/files#diff-768887e6132bb8207503a7a5d74d02de19f165eb043b82853b3a8192b3e26c06R350

About calling typeof once more, that's my question, is it an extra cost?

Reply:
On second thought, redundant calling typeof might not be wise. Hermes is not optimized for repeated actions like e.g. a V8. I'd keep it in your proposed version then.